lib/install/vchlog/buffered_pipe_test.go

```diff
@@ -205,6 +205,41 @@ func TestPipeWriteClose(t *testing.T) {
 	assert.Equal(t, 0, n, "write on closed pipe returned %d bytes", n)
 }
 
+// Test read left-over data after pipe is closed
+func TestPipeReadAfterWriteClose(t *testing.T) {
+	testByte := []byte("hello world")
+	toRead := make([]byte, 64)
+	bp := NewBufferedPipe()
+	writeDone := make(chan int)
+
+	go write(t, bp, testByte, len(testByte), writeDone)
+	<-writeDone
+	go bp.Close()
+
+	// wait for write end to close
+	// closing of read end is expected to block until data is drained from the pipe
+	time.Sleep(1 * time.Millisecond)
```

Comment:
Both here and below should really be `until` logic - for simplicity the global test timeout can be used to avoid something here (I did not see a handy testify until mechanism).
As it stands it will probably work, until run on a heavily loaded system and 1ms is not long enough for the other goroutine to be scheduled - for example if running unit tests in parallel.

Reply:
Thanks for pointing that out. I think there's a better way around to fix this: instead of waiting or timing out, it could continuously write to the pipe until the write fails, while the close() is running in the background in a go routine. Then, after the write has failed, begin reading on the left-over data until exhausted. I can keep a count of the number of bytes written before failing and check if the read fails after reading the exact number of bytes. Fixes coming up.